Extension/src/LanguageServer/customProviders.ts

```diff
@@ -135,6 +136,11 @@ export class CustomConfigurationProviderCollection {
     }
 
     public add(provider: CustomConfigurationProvider, version: Version): boolean {
+        if (new CppSettings().intelliSenseEngine === "Disabled") {
```

Comment:
Should the user of this function already have determined whether IntelliSense is disabled before preparing their `CustomConfigurationProvider`? If that is the case, perhaps an exception would be more apt

Reply:
Even if they should, it doesn't mean they will, so we need to protect against it.

Reply:
I think I had a different context in mind when responding.  The sole caller of this function is the `registerCustomConfigurationProvider` method which is invoked by the provider.  I don't think we should expect the providers to check our settings before registering, but I do think that we should protect against adding providers to the collection and since this is the method that does that, this is where the check should be made (rather than requiring every potential caller to know that they have to make this check(